pingcap/tispark
# Move the dependency of API v1 in Catalyst

### What problem does this PR solve? 
https://github.com/pingcap/tispark/issues/2242

### What is changed and how it works?
- Add TiDBTable
- Add TiDBTableProvider and delete TiDBDataSource
- Replace `TiDBRelation` to `TiDBTable`
- Delete some catalyst extension because of useless now
- fix some test
- change code structure

### Check List 

Tests 

 - Unit test
 - Integration test

Side effects

The following usage is no longer supported, because is a datasource v1 read/write path
```
sprak.sql("use spark_catalog")  
spark.sql("create table xxx using tidb")
spark.sql("select * from xxx") 
spark.sql("insert into xxx")  
```
- use tidb_catalog to select instead
- use write api to insert instead
